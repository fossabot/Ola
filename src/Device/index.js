import React from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'
import Spinner from '../Spinner'


const Device = ({variant, url, className,busy,...props})=> {
  const styles = cx(
    'ola_device',
    { [`ola_device-${variant}`]: variant },
    { 'is-busy': busy },
    className
  )
 
  return (
    <div className={styles}{...props}>
      <div className='ola_device-frame'>
        {url ? <img
          className={'ola_device-content'}
          src={url} /> : busy &&
          <div className={'ola_device-content ola_device-spinner'}>
            <Spinner size="medium"/>
          </div>}
      </div>
      <div className="ola_device-stripe"></div>
      <div className="ola_device-header"></div>
      <div className="ola_device-sensors"></div>
      <div className="ola_device-btns"></div>
      <div className="ola_device-power"></div>
    </div>
  )
}
Device.defaultProps = {
  className: null,
  busy: null,
  variant: 'iphone-8',
}

Device.propTypes = {
  /** Device variants */
  variant: PT.oneOf(['iphone-8', 'iphone-x']),
  /** Extra className */
  className: PT.string,
  /** Url img */
  url: PT.string,
  /** Loading */
  busy: PT.oneOfType([
    PT.string,
    PT.bool
  ]),
}

export default Device
