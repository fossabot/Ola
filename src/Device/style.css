
.ola_device *,
.ola_device ::after,
.ola_device ::before,
.ola_device::after,
.ola_device::before {
  box-sizing: border-box;
  display: block;
}
.ola_device {
  --scale: 0.5;
  position: relative;
  /* @media (--from-width-2){
    --scale: 0.3;
  } */
  z-index: 1;
}

.ola_device .ola_device-frame {
    z-index: 1;
  }
  .ola_device .ola_device-content {
    background-color: #fff;
    background-position: center center;
    background-size: cover;
    object-fit: cover;
    position: relative;
  }
  
  .ola_device-iphone-8 {
      width: max-content;
  }

  
  .ola_device-iphone-8 .ola_device-frame {
    background: #fff;
    border-radius: calc(68px * var(--scale));
    box-shadow: inset 0 0 0 calc(2px * var(--scale)) #c8cacb,
      inset 0 0 0 calc(7px * var(--scale)) #e2e3e4;
    height: calc(871px * var(--scale));
    padding: calc(102px * var(--scale)) calc(22px * var(--scale));
    width: calc(419px * var(--scale));
  }
    .ola_device-spinner {
    /* background-color: var(--gray-xlight); */
    width: 100%;
    height: 100%;
    }
    svg{
        color: var(--brand)
    }
  .ola_device-iphone-8 .ola_device-content {
    border: 2px solid #222;
    border-radius: 4px;
    height: calc(667px * var(--scale));
    width: calc(375px * var(--scale));
  }
  .ola_device-iphone-8 .ola_device-stripe::after,
  .ola_device-iphone-8 .ola_device-stripe::before {
    border: solid rgba(51, 51, 51, 0.15);
    border-width: 0 calc(7px * var(--scale));
    content: "";
    height: calc(6px * var(--scale));
    left: 0;
    position: absolute;
    width: 100%;
    z-index: 9;
  }
  .ola_device-iphone-8 .ola_device-stripe::after {
    top: calc(68px * var(--scale));
  }
  .ola_device-iphone-8 .ola_device-stripe::before {
    bottom: calc(68px * var(--scale));
  }
  .ola_device-iphone-8 .ola_device-header {
    border: calc(2px * var(--scale)) solid #c8cacb;
    border-radius: 50%;
    bottom: calc(25px * var(--scale));
    height: calc(58px * var(--scale));
    left: 50%;
    margin-left: calc(-29px * var(--scale));
    position: absolute;
    width: calc(58px * var(--scale));
  }
  .ola_device-iphone-8 .ola_device-sensors {
    background: #666;
    border-radius: 3px;
    height: calc(6px * var(--scale));
    left: 50%;
    margin-left: calc(-38px * var(--scale));
    position: absolute;
    top: calc(52px * var(--scale));
    width: calc(76px * var(--scale));
  }
  .ola_device-iphone-8 .ola_device-sensors::after,
  .ola_device-iphone-8 .ola_device-sensors::before {
    background: #666;
    border-radius: 50%;
    content: "";
    position: absolute;
  }
  .ola_device-iphone-8 .ola_device-sensors::after {
    height: calc(10px * var(--scale));
    left: calc(50% * var(--scale));
    margin-left: -5px;
    top: calc(-25px * var(--scale));
    width: calc(10px * var(--scale));
  }
  .ola_device-sensors::before {
    height: calc(12px * var(--scale));
    left: calc(-42px * var(--scale));
    margin-top: -6px;
    top: 50%;
    width: calc(12px * var(--scale));
  }
  .ola_device-btns {
    background: #c8cacb;
    height: calc(30px * var(--scale));
    left: calc(-3px * var(--scale));
    position: absolute;
    top: calc(102px * var(--scale));
    width: calc(3px * var(--scale));
  }
  .ola_device-btns::after,
  .ola_device-btns::before {
    background: #c8cacb;
    content: "";
    height: calc(56px * var(--scale));
    left: 0;
    position: absolute;
    width: calc(3px * var(--scale));
  }
  .ola_device-btns::after {
    top: calc(62px * var(--scale));
  }
  .ola_device-btns::before {
    top: calc(132px * var(--scale));
  }
  .ola_device-power {
    background: #c8cacb;
    height: calc(80px * var(--scale));
    position: absolute;
    right: calc(-2 * var(--scale));
    top: 160px;
    width: calc(3px * var(--scale));
  }
  .ola_device-iphone-8.ola_device-gold .ola_device-frame {
    box-shadow: inset 0 0 0 calc(2px * var(--scale)) #c8cacb,
      inset 0 0 0 calc(7px * var(--scale)) #e2e3e4;
  }
  .ola_device-iphone-8.ola_device-gold .ola_device-header {
    border-color: #e4b08a;
  }
  .ola_device-iphone-8.ola_device-gold .ola_device-btns,
  .ola_device-iphone-8.ola_device-gold .ola_device-btns::after,
  .ola_device-iphone-8.ola_device-gold .ola_device-btns::before {
    background: #e4b08a;
  }
  .ola_device-iphone-8.ola_device-gold .ola_device-power {
    background: #e4b08a;
  }
  .ola_device-iphone-8.ola_device-spacegray .ola_device-frame {
    background: #222;
    box-shadow: inset 0 0 0 2px #74747a, inset 0 0 0 7px #9b9ba0;
  }
  .ola_device-iphone-8.ola_device-spacegray .ola_device-stripe::after,
  .ola_device-iphone-8.ola_device-spacegray .ola_device-stripe::before {
    border-color: rgba(204, 204, 204, 0.35);
  }
  .ola_device-iphone-8.ola_device-spacegray .ola_device-btns,
  .ola_device-iphone-8.ola_device-spacegray .ola_device-btns::after,
  .ola_device-iphone-8.ola_device-spacegray .ola_device-btns::before {
    background: #74747a;
  }
  

/* iphoneX */


  .ola_device-iphone-x {
    height: 868px;
    width: 428px;
  }
  
  .ola_device-iphone-x .ola_device-frame {
    background: #222;
    border-radius: 68px;
    box-shadow: inset 0 0 2px 2px #c8cacb, inset 0 0 0 7px #e2e3e4;
    height: 868px;
    padding: 28px;
    width: 428px;
  }
  
  .ola_device-iphone-x .ola_device-content {
    border-radius: 40px;
    height: 812px;
    width: 375px;
  }
  
  .ola_device-iphone-x .ola_device-stripe::after,
  .ola_device-iphone-x .ola_device-stripe::before {
    border: solid rgba(51, 51, 51, .25);
    border-width: 0 7px;
    content: "";
    height: 7px;
    left: 0;
    position: absolute;
    width: 100%;
    z-index: 9;
  }
  
  .ola_device-iphone-x .ola_device-stripe::after {
    top: 85px;
  }
  
  .ola_device-iphone-x .ola_device-stripe::before {
    bottom: 85px;
  }
  
  .ola_device-iphone-x .ola_device-header {
    background: #222;
    border-bottom-left-radius: 20px;
    border-bottom-right-radius: 20px;
    height: 30px;
    left: 50%;
    margin-left: -102px;
    position: absolute;
    top: 28px;
    width: 204px;
  }
  
  .ola_device-iphone-x .ola_device-header::after,
  .ola_device-iphone-x .ola_device-header::before {
    content: "";
    height: 10px;
    position: absolute;
    top: 0;
    width: 10px;
  }
  
  .ola_device-iphone-x .ola_device-header::after {
    background: radial-gradient(circle at bottom left, transparent 0, transparent 75%, #222 75%, #222 100%);
    left: -10px;
  }
  
  .ola_device-iphone-x .ola_device-header::before {
    background: radial-gradient(circle at bottom right, transparent 0, transparent 75%, #222 75%, #222 100%);
    right: -10px;
  }
  
  .ola_device-iphone-x .ola_device-sensors::after,
  .ola_device-iphone-x .ola_device-sensors::before {
    content: "";
    position: absolute;
  }
  
  .ola_device-iphone-x .ola_device-sensors::after {
    background: #444;
    border-radius: 3px;
    height: 6px;
    left: 50%;
    margin-left: -25px;
    top: 32px;
    width: 50px;
  }
  
  .ola_device-iphone-x .ola_device-sensors::before {
    background: #444;
    border-radius: 50%;
    height: 14px;
    left: 50%;
    margin-left: 40px;
    top: 28px;
    width: 14px;
  }
  
  .ola_device-iphone-x .ola_device-btns {
    background: #c8cacb;
    height: 32px;
    left: -3px;
    position: absolute;
    top: 115px;
    width: 3px;
  }
  
  .ola_device-iphone-x .ola_device-btns::after,
  .ola_device-iphone-x .ola_device-btns::before {
    background: #c8cacb;
    content: "";
    height: 62px;
    left: 0;
    position: absolute;
    width: 3px;
  }
  
  .ola_device-iphone-x .ola_device-btns::after {
    top: 60px;
  }
  
  .ola_device-iphone-x .ola_device-btns::before {
    top: 140px;
  }
  
  .ola_device-iphone-x .ola_device-power {
    background: #c8cacb;
    height: 100px;
    position: absolute;
    right: -3px;
    top: 200px;
    width: 3px;
  }


  