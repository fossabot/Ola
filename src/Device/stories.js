import React from 'react'
import Device from './'

export default {
  title: 'Device',
  component: Device,
  args: {
    variant: '',
    url:'',
    busy: null
  }
}

export const Base = (args) => <Device  {...args}/>

