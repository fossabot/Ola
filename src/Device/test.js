import React from 'react'
import Device from './'
import renderer from 'react-test-renderer'


it('Default Device', () => {
  const tree = renderer.create(<Device/>).toJSON() 
  expect(tree).toMatchSnapshot()
})

it('Device Busy', () => {
  const tree = renderer
    .create(<Device busy="Test"/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Device with url', () => {
  const tree = renderer
    .create(<Device url={'https://via.placeholder.com/150'} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})