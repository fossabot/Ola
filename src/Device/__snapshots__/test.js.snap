// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Device 1`] = `
<div
  className="ola_device ola_device-iphone-8"
>
  <div
    className="ola_device-frame"
  />
  <div
    className="ola_device-stripe"
  />
  <div
    className="ola_device-header"
  />
  <div
    className="ola_device-sensors"
  />
  <div
    className="ola_device-btns"
  />
  <div
    className="ola_device-power"
  />
</div>
`;

exports[`Device Busy 1`] = `
<div
  className="ola_device ola_device-iphone-8 is-busy"
>
  <div
    className="ola_device-frame"
  >
    <div
      className="ola_device-content ola_device-spinner"
    >
      <svg
        className="ola_spinner is-medium"
        height={76}
        viewBox="0 0 76 76"
        width={76}
      >
        <circle
          cx={38}
          cy={38}
          r={34}
        />
      </svg>
    </div>
  </div>
  <div
    className="ola_device-stripe"
  />
  <div
    className="ola_device-header"
  />
  <div
    className="ola_device-sensors"
  />
  <div
    className="ola_device-btns"
  />
  <div
    className="ola_device-power"
  />
</div>
`;

exports[`Device with url 1`] = `
<div
  className="ola_device ola_device-iphone-8"
>
  <div
    className="ola_device-frame"
  >
    <img
      className="ola_device-content"
      src="https://via.placeholder.com/150"
    />
  </div>
  <div
    className="ola_device-stripe"
  />
  <div
    className="ola_device-header"
  />
  <div
    className="ola_device-sensors"
  />
  <div
    className="ola_device-btns"
  />
  <div
    className="ola_device-power"
  />
</div>
`;
